src/DynamoCoreWpf/ViewModels/Search/NodeAutoCompleteSearchViewModel.cs

```diff
@@ -155,67 +163,224 @@ private void InitializeDefaultAutoCompleteCandidates()
 
         internal void DisplayMachineLearningResults()
         {
-            // Case 1: no results (0 items)
-            FilteredResults = new List<NodeSearchElementViewModel>();
-            DisplayNoRecommendationsLowConfidence = true;
-            DisplayLowConfidence = false;
-            NoRecommendationsOrLowConfidenceTitle = Resources.AutocompleteNoRecommendationsTitle;
-            NoRecommendationsOrLowConfidenceMessage = Resources.AutocompleteNoRecommendationsMessage;
+            // Intialize request for the the ML API
+            MLNodeAutoCompletionRequest request = new MLNodeAutoCompletionRequest();
+
+            var nodeInfo = PortViewModel.PortModel.Owner;
+            var portInfo = PortViewModel.PortModel;
+
+            request.DynamoVersion = AssemblyHelper.GetDynamoVersion().ToString();
+            request.NumberOfResults = 10;
+
+            // Set node info
+            request.Node.Id = nodeInfo.GUID.ToString();
+            request.Node.Lacing = nodeInfo.ArgumentLacing.ToString();
 
-            // Case 2: the result has at least one item assuming each node could be by recommendation or by use
-            /*
-            FilteredResults = DefaultResults.Where(e => e.Name == "Watch 3D" || e.Name == "Python Script").ToList();
-            foreach (var item in FilteredResults)
+            if (nodeInfo is DSFunctionBase functionNode)
             {
-                item.Model.AutoCompletionNodeMachineLearningInfo.ViewConfidenceScoreRecentUse = true;
-                item.Model.AutoCompletionNodeMachineLearningInfo.IsByUse = true;
+                request.Node.Type.Id = functionNode.CreationName;
+            }
+            else if (nodeInfo is NodeModel nodeModel)
+            {
+                string typeId = GenerateTypeIdForNodeModelNode(nodeModel.GetType().AssemblyQualifiedName);
+                request.Node.Type.Id = typeId;
             }
-            DisplayNoRecommendationsLowConfidence = !FilteredResults.Where(n => n.Model.AutoCompletionNodeMachineLearningInfo.IsByRecommendation).Any();
-            DisplayLowConfidence = false;
-            NoRecommendationsOrLowConfidenceTitle = Resources.AutocompleteNoRecommendationsTitle;
-            NoRecommendationsOrLowConfidenceMessage = Resources.AutocompleteNoRecommendationsMessage;
-            */
 
+            // Set port info
+            request.Port.Direction = portInfo.PortType == PortType.Input ? PortType.Input.ToString().ToLower() : PortType.Output.ToString().ToLower();
+            request.Port.Name = portInfo.Name;
```

Comment:
Nice catch. So just to to clear for the variable input nodes, should the port-name be empty only for the ports that are additionally added by user or for all the ports of that node? This is because the inbuilt ports for those nodes can be of different types but the additional ones that can be added will be of same type.

Reply:
After discussing with Martin and Varvara, we will be parsing the port name to remove the digits at the end.